

*   nama: la liani
*   program : FTDS18-rmt
*   ptojek : LC2



## Problem 1




Kamu sedang mencari rumah dan akan memilih satu dari dua pilihan rumah dengan harga yang cukup berbeda. Sekilas, kedua rumah sama karakteristiknya, namun kamu ingin memastikan kembali apakah rumah yang lebih mahal patut dibeli atau tidak. Berikut data kedua rumah:

###import package yang diperlukan 

In [11]:
import numpy as np
import pandas as pd

membuat vektor rumah_1 dan rumah_2 menggunakan numpy

In [3]:
vektor_rumah_1 = np.array([100, 240, 3,3,2,0,1,1]) 
vektor_rumah_2 = np.array([120,100,2,2,2,1,1,1])

menghitung cosine similarity dengan rumus:
cos tehtha = a.b/norm a.norm b

In [4]:
dot = np.dot(vektor_rumah_1,vektor_rumah_2)
norm_vektor_rumah_1 = np.linalg.norm(vektor_rumah_1)
norm_vektor_rumah_2 = np.linalg.norm(vektor_rumah_2)

In [22]:
dot

36018

In [23]:
norm_vektor_rumah_1*norm_vektor_rumah_2

40632.99102945783

In [10]:
cosine_similarity = dot/(norm_vektor_rumah_1*norm_vektor_rumah_2)
cosine_similarity

0.8864225617525403

dapat disimpulkan kedua rumah tersebut cukup mirip karena nilai cosine similarity nya adalah 0.8864,karna sudut antara kedua vektor sangatlah kecil hampir mendekati nol. saya akan memilih rumah_2 karena harganya jauh labih murah dibanding rumah_1

# Problem 2

Kamu diberikan tugas untuk menganalisis harga sewa apartemen di Argentina per Januari 2015. Gunakan kemampuan statistikamu untuk menganalisis data yang tersedia!

Dataset Description
Pada graded challenge ini, data diakses menggunakan bigquery-public-data pada Google Cloud Big Query.
Buka Google Cloud Platform, masuk ke BigQuery, lalu buka tab bigquery-public-data atau klik link berikut atau link berikut untuk langsung menuju ke tabel.

hanya kolom price yang akan kita analisis karena kolom fo dan f1 yang memuat true dan false adalah hanya menunjukan bahwa pada kolom price akan diambil ketika element  kolom opertion dan type_property adalah "rent dan "Apartement.

 queri yang digunakan adalah SELECT price, operation = "rent", property_type = "Apartement" FROM `bigquery-public-data.properati_properties_ar.properties_rent_201501` LIMIT 1000

In [13]:
data = pd.read_csv("/content/drive/MyDrive/bq-results-20230213-035143-1676260360074/h8dsft_P0LC3_<la_liani>.csv")
data

price   f0_    f1_
0     2500  True  False
1     3500  True  False
2     2850  True  False
3     1850  True  False
4     7500  True  False
..     ...   ...    ...
995  18000  True  False
996   4000  True  False
997   5500  True  False
998   3200  True  False
999   3100  True  False

[1000 rows x 3 columns]

In [18]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   price   1000 non-null   int64
 1   f0_     1000 non-null   bool 
 2   f1_     1000 non-null   bool 
dtypes: bool(2), int64(1)
memory usage: 9.9 KB


data tidak memiliki missing value

In [14]:
data.head()

price   f0_    f1_
0   2500  True  False
1   3500  True  False
2   2850  True  False
3   1850  True  False
4   7500  True  False

In [15]:
data.tail()

price   f0_    f1_
995  18000  True  False
996   4000  True  False
997   5500  True  False
998   3200  True  False
999   3100  True  False

  ###### Clue 
Untuk mengetahui adanya anomali, kamu bisa menggunakan metode extreme value analysis. Untuk melakukan pengecekan anomali/outlier, lakukan langkah-langkah di bawah ini:

Lakukan perhitungan central tendency (mean, median, modus) terhadap data sebelum dideteksi adanya anomali.
Cek skewness data untuk mengetahui apakah data terdistribusi normal atau tidak.
Lakukan pengolahan data dengan menggunakan extreme value analysis.
Buat variabel baru yang menyimpan data yang sudah dibuang data anomalinya.

In [17]:
print ('STD: ' + str(data['price'].std())) # mecari nilai aggregate dengan fungsi aggregate
print ('Min: ' + str(data['price'].min()))
print ('Mode: ' + str(data['price'].mode()[0]))
print ('Median: ' + str(data['price'].median()))
print ('Mean: ' + str(data['price'].mean().round(2)))
print ('Max: ' + str(data['price'].max()))

STD: 37289.59328356529
Min: 500
Mode: 3500
Median: 6000.0
Mean: 15207.26
Max: 585000


PERTANYAAN

Silahkan jawab pertanyaan-pertanyaan di bawah ini berdasarkan hasil yang kamu peroleh dan tulis pada bagian hasil:

1. Berapa rata-rata, median, dan modus dari data tersebut sebelum dihilangkan outliernya? 
2. Bagaimana kecerendungan pemusatan datanya? jelaskan jawabanmu!
Sebelum melakukan extreme value analysis, kamu harus melakukan pengecekan skewness dari distribusi datanya. Apakah datanya skew atau normal? jelaskan jawabanmu!
3. Ada dua teknik untuk melakukan extreme value analysis, teknik yang mana yang kamu pakai? berikan alasanmu berdasarkan data!

jawaban 1. rata rata, median, dan modus data price diatas adalah berturut turut 15207.26, 6000.00, 3500

jawaban 2. untuk mengetahui nilai skewness maka kita gunakan rumus :
rata-rata - modus/ standar deviasi

In [19]:
mean = 15207.26 
modus = 3500
std = 37289.59328356529

In [21]:
skewnees = (mean - modus)/std
skewnees

0.3139551539479987

dapat dilihat bahwa nilai skewnees = 0.3139551539479987 yakni diantara -0.5 dan 0.5 maka data price masih dianggap normal atau terdistribusi normal.

jawabn 3. uji yang saya lakukan yakni dengan menggunakan box dan whisker plot. karena datanya menggunakan uji nilai skewness tidak ada anomali sehingga tidak membuat kolom baru untuk menyimpan data sesudah dibuang outliernya. 